lib/internal/process/execution.js

```diff
@@ -159,6 +162,7 @@ function createOnGlobalUncaughtException() {
     }
 
     const type = fromPromise ? 'unhandledRejection' : 'uncaughtException';
+    process.emit(kUncaughtExceptionMonitor, er, type);
```

Comment:
Could this end up in an infinite loop in case the listener functions throws sync?

Reply:
I will try

Reply:
This results in no endless loop. In then end it's the same as a sync throw in a regular uncaughtException listener.
Clearly the regular handlers get not invoked if this happens but this is mostly the same for regular uncaughtException listeners if e.g. the first throws.